### PR TITLE
Rule endopoints

### PR DESCRIPTION
**Pull request by leosoubeste**

### CODE REVIEW

File: lib/auth0/api/v2/rules.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          }
+          path = '/api/v2/rules'
+          get(path, request_params)
+        end
+
+        alias_method :get_rules, :rules
+
+        # https://auth0.com/docs/api/v2#!/Rules/get_rules_by_id
+        def rule(rule_id, options = {})
+          fail Auth0::InvalidParameter, 'Must supply a valid rule id' if rule_id.to_s.empty?
```
Comment (falvarisqueta) [inline lib/auth0/api/v2/rules.rb:22]: would unless rule_id work? (Question applies to all Params)

File: spec/integration/lib/auth0/api/v2/api_rules_spec.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    script = 'function (user, context, callback) { callback(null, user, context);}'
+    stage = 'login_success'
+    @enabled_rule = client.create_rule("Enabled Rule #{suffix}", script, rand(1..10), true, stage)
+    @disabled_rule = client.create_rule("Disabled Rule #{suffix}", script, rand(11..20), false, stage)
+  end
+
+  after(:all) do
+    rules = client.rules
+    rules.each do |rule|
+      client.delete_rule(rule['id'])
```
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_rules_spec.rb:17]: Does it make sense to have a clean_rules?

File: spec/integration/lib/auth0/api/v2/api_rules_spec.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+      it do
+        expect(client.rules(enabled: false).size).to be 1
+      end
+
+      it do
+        expect(
+          client.rules(
+            enabled: true,
+            fields: [:script, :order].join(',')).first).to(include('script', 'order'))
```
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_rules_spec.rb]: we could move include to a separate line to be consistent with line 46 of same file

File: spec/lib/auth0/api/v2/rules_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    end
+  end
+
+  context '.create_rule' do
+    it { expect(@instance).to respond_to(:create_rule) }
+    it 'is expected to call post /api/v2/rules' do
+      expect(@instance).to receive(:post).with(
+        '/api/v2/rules',
+        name: 'test', script: 'script', order: 'order', enabled: false, stage: 'login_success')
+      expect do
```
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/rules_spec.rb]: Why do end block here? Line 24 does the same using {}

File: spec/lib/auth0/api/v2/rules_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        @instance.create_rule('test', 'script', 'order', false)
+      end.not_to raise_error
+    end
+  end
+  context '.update_rule' do
+    it { expect(@instance).to respond_to(:update_rule) }
+    it 'is expected to call put /api/v2/rules/test' do
+      expect(@instance).to receive(:patch).with(
+        '/api/v2/rules/test', script: 'script', order: 'order', enabled: true, stage: 'some_stage')
+      expect do
```
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/rules_spec.rb:51]: Why do end block here? Line 53 does the same using {}

File: spec/integration/lib/auth0/api/v2/api_rules_spec.rb (hunk 3 above)
Comment (leosoubeste) [inline spec/integration/lib/auth0/api/v2/api_rules_spec.rb]: moved both to one line

File: spec/lib/auth0/api/v2/rules_spec.rb (hunk 4 above)
Comment (leosoubeste) [inline spec/lib/auth0/api/v2/rules_spec.rb]: moved to one line

File: spec/lib/auth0/api/v2/rules_spec.rb (hunk 5 above)
Comment (leosoubeste) [inline spec/lib/auth0/api/v2/rules_spec.rb:51]: Rubocop does not allow to have a multiline block with {}, it has to be moved to do-end. In this case, line is greater that 120 chars, so we keep the do-end.

File: lib/auth0/api/v2/rules.rb (hunk 1 above)
Comment (leosoubeste) [inline lib/auth0/api/v2/rules.rb:22]: It is easier for me to read "if empty" on validations.

File: spec/integration/lib/auth0/api/v2/api_rules_spec.rb (hunk 2 above)
Comment (leosoubeste) [inline spec/integration/lib/auth0/api/v2/api_rules_spec.rb:17]: It is the only method on the "after" clause. Does not make sense at this time.

File: lib/auth0/api/v2/rules.rb (hunk 1 above)
Comment (ignaciojonas) [inline lib/auth0/api/v2/rules.rb:22]: I would agree with @falvarisqueta. Tomorrow we can discuss about this.

File: spec/lib/auth0/api/v2/rules_spec.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
```
Comment (ignaciojonas) [inline spec/lib/auth0/api/v2/rules_spec.rb]: We can add some tests to expect the exception.

File: lib/auth0/api/v2/rules.rb (hunk 1 above)
Comment (ignaciojonas) [inline lib/auth0/api/v2/rules.rb:22]: We agreed to keep the if empty, because covers more scenarios.

Comment (falvarisqueta) [inline lib/auth0/api/v2/rules.rb:22]: Ok

File: spec/lib/auth0/api/v2/rules_spec.rb (hunk 5 above)
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/rules_spec.rb:51]: Ok

File: spec/integration/lib/auth0/api/v2/api_rules_spec.rb (hunk 2 above)
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_rules_spec.rb:17]: Ok

**Comment by falvarisqueta** [comment]
Looks good to me

**Pull request merged**